core/src/main/scala/filodb.core/downsample/DownsampledTimeSeriesShard.scala

```diff
@@ -234,6 +234,11 @@ class DownsampledTimeSeriesShard(rawDatasetRef: DatasetRef,
           // is expensive, but we do it to handle data sizing for metrics that have
           // continuous churn. See capDataScannedPerShardCheck method.
           val recs = partKeyIndex.partKeyRecordsFromFilters(filters, chunkMethod.startTime, chunkMethod.endTime)
+          if (recs.size > 100000) {
+            // if we have a 3 bit spread and one shard has more than 100K tss, it means
+            // that the cardinality of the query is ~800K close to a million, hence we want to log such queries
+            logger.warn(s"Number of tss: ${recs.size} for query ${querySession.qContext.origQueryParams}")
```

Comment:
@kvpetrov is it also possible to add a toString override here https://github.com/filodb/FiloDB/blob/develop/core/src/main/scala/filodb.core/query/QueryContext.scala#L17 and convert it to a key=value pairs for better readability in logs for `querySession.qContext.origQueryParams`?